File: JW/CodeTest/OT_001.py
# board = [[0, 0, 0, 0, 0],
#          [0, 0, 1, 0, 3],
#          [0, 2, 5, 0, 1],
#          [4, 2, 4, 4, 2],
#          [3, 5, 1, 3, 1]]

# moves = [1, 5, 3, 5, 1, 2, 1, 4]


def solution(board, moves):
    answer = 0
    stack = []
    for i in range(len(moves)):

        grapCrane(board, moves[i] - 1, stack)
        if len(stack) > 1 and popingDoll(stack):
            answer += 2
    print(stack)
    return answer


def grapCrane(board, move, arr):

    for doll in board:
        if doll[move] > 0:
            arr.append(doll[move])
            doll[move] = 0
            break;


def popingDoll(stack):
    if stack[len(stack)-1] == stack[len(stack)-2]:
        stack.pop()
        stack.pop()
        return True
    else:
        return False
    


# solution(board, moves)

File: JW/CodeTest/OT_001.js
function solution(board, moves) {
	let bucket = []
	let answer = 0

	moves.forEach((move) => {
		for (let i = 0; i < board.length; i++) {
			const val = board[i][move - 1]

			// 뽑음
			if (val > 0) {
				// 뽑았으니 board값 0으로 변경
				board[i][move - 1] = 0

				// 비어 있을 때
				if (bucket.length === 0) {
					bucket = [ val ]
				} else if (bucket[bucket.length - 1] === val) {
					bucket.splice(bucket.length - 1, 1) // 마지막 인자 제거
					answer = answer + 2
				} else {
					// 바구니 마지막 값이랑 다르면 담자.
					bucket = [ ...bucket, val ]
				}

				break
			}
		}
	})

	return answer
}